src/bindings/python/src/pyopenvino/frontend/graph_iterator.hpp

```diff
@@ -0,0 +1,18 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <pybind11/pybind11.h>
+
+#include "openvino/frontend/graph_iterator.hpp"
+
+namespace py = pybind11;
+
+class PyIGraphIterator : public ov::frontend::IGraphIterator {
+public:
+    using IGraphIterator::IGraphIterator; // Inherit constructors
+};
```

Comment:
can we avoid introducing `GraphIterator` class in common frontend? That is because `GraphIterator` is not used by other frontends and we are going to remove it in the future.

Reply:
This class serves as common predecessor for Python and C++ `GraphIterator`. This predecessor should have both Python and c++ implementation. 
Initially I tried to use c++ `GraphIterator` as such common predecessor, but there was a problem with creating a Python binding for it. As far as I remember pyopenvino does not see classes which are not in common frontend so there was impossible to make a binding, as `GraphIterator` could not be imported.
So I needed the predecessor to be in common frontend, otherwise the binding could not be created.

Reply:
If we remove GraphIterator in future, we will need to change architecture of tf.Graph decoder, because currently it expects GraphIterator + node Decoder pipeline.